pysat/utils/io.py

```diff
@@ -0,0 +1,1159 @@
+                        try:
+                            # Start by assuming it is a DataFrame
+                            iterable = inst[key].iloc[0].columns
+                            is_frame = True
+                        except AttributeError:
+                            # Otherwise get sub-variables for a Series
+                            iterable = [inst[key].iloc[0].name]
+                            is_frame = False
+
+                        # Find the subvariable data within the main variable,
+                        # checking that this is not an empty DataFrame or
+                        # Series. Determine the underlying data types
+                        good_data_loc = 0
+                        for idat in np.arange(len(inst.data)):
+                            if len(inst.data[key].iloc[0]) > 0:
+                                data_loc = idat
+                                break
+
+                        # Found a place with data, if there is one
+                        # now iterate over the subvariables, get data info
+                        # create netCDF4 variables and store the data
+                        # stored name is variable_subvariable
+                        for col in iterable:
+                            if is_frame:
+                                # We are working with a DataFrame, so
+                                # multiple subvariables stored under a single
+                                # main variable heading
+                                idx = inst[key].iloc[good_data_loc][col]
+                                data, coltype, _ = inst._get_data_info(idx)
+                                cdfkey = out_data.createVariable(
+                                    '_'.join((case_key, col)), coltype,
+                                    dimensions=var_dim, zlib=zlib,
+                                    complevel=complevel, shuffle=shuffle)
+
+                                lkey = '_'.join((case_key, col))
+                                if lkey in export_meta.keys():
+                                    cdfkey.setncatts(filter_netcdf4_metadata(
+                                        inst, export_meta[lkey], coltype,
+                                        export_nan=export_nan))
+                                else:
+                                    pysat.logger.info(
+                                        ''.join(('Unable to find MetaData for ',
+                                                 lkey)))
+
+                                # Attach data.  It may be slow to repeatedly
+                                # call the store method as well astype method
+                                # below collect data into a numpy array, then
+                                # write the full array in one go
+                                temp_cdf_data = np.zeros(
+                                    (num, dims[0]), dtype=coltype)
```

Comment:
```suggestion
                                temp_cdf_data = np.zeros(
                                    (num, dims[0])).astype(coltype)
```
Not sure why, but using dtype only allows one-character strings to be put in the array.  This is the source of 5 out of 6 of the pandas 25 bugs.  🐼 🐛

Reply:
If coltype=='str', we could specify `dtype='U32'` to maintain behavior but eliminate `astype`.

But 2D pandas is going to be deprecated soon, so optimization of these codes is probably not high priority.